flang/lib/Optimizer/Transforms/SimplifyRegionLite.cpp

```diff
@@ -0,0 +1,46 @@
+//===- SimplifyRegionLite.cpp -- region simplification lite ---------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "PassDetail.h"
+#include "flang/Optimizer/Dialect/FIROps.h"
+#include "flang/Optimizer/Transforms/Passes.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/DialectConversion.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+#include "mlir/Transforms/RegionUtils.h"
+
+namespace {
+
+class SimplifyRegionLitePass
+    : public fir::SimplifyRegionLiteBase<SimplifyRegionLitePass> {
+public:
+  void runOnOperation() override;
+};
+
+class DummyRewriter : public PatternRewriter {
+  public:
+    DummyRewriter(MLIRContext *ctx) : PatternRewriter(ctx) {}
+};
+
+} // namespace
+
+void SimplifyRegionLitePass::runOnOperation() {
+  auto op = getOperation();
+  auto regions = op->getRegions();
+  mlir::RewritePatternSet patterns(op.getContext());
+  DummyRewriter rewriter(op.getContext());
+  if (regions.empty())
+    return;
+
+  (void)mlir::eraseUnreachableBlocks(rewriter, regions);
+  (void)mlir::runRegionDCE(rewriter, regions);
+}
+
+std::unique_ptr<mlir::Pass> fir::createSimplifyRegionLitePass() {
```

Comment:
I think we may be running simplifyRegions straight out of the bridge. Yeah, we are. https://github.com/flang-compiler/f18-llvm-project/blob/fir-dev/flang/lib/Lower/Bridge.cpp#L2503-L2505 This might be a good point to move that out of the bridge, actually.